universe/Planet.cpp

```diff
@@ -871,3 +871,23 @@ void Planet::ClampMeters() {
     UniverseObject::GetMeter(METER_REBEL_TROOPS)->ClampCurrentToRange();
     UniverseObject::GetMeter(METER_DETECTION)->ClampCurrentToRange();
 }
+
+const std::string Planet::OrdinalSuffix() const {
+    TemporaryPtr<const System> system_ptr = GetSystem(this->SystemID());
+    int orbit = system_ptr->OrbitOfPlanet(this->ID());
+    if (m_size == SZ_ASTEROIDS) {  // asteroids are named according to orbit position
+        if (orbit < 1) {
+            orbit = system_ptr->NumPlanets();
+        }
+        return UserString("NEW_ASTEROID_SUFFIX_ORBIT_" + RomanNumber(orbit));
+    } else {
+        if (orbit < 1) {
+            const std::set<int>& system_planet_ids = system_ptr->PlanetIDs();
+            for (std::set<int>::const_iterator it = system_planet_ids.begin(); it != system_planet_ids.end(); ++it) {
+                if (GetPlanet(*it)->Size() != SZ_ASTEROIDS)
```

Comment:
I'd add a safety check on the result of GetPlanet before dereferencing the pointer...